spec-tests/runners/kzg.rs

```diff
@@ -0,0 +1,341 @@
+        }
+    };
+
+    let result = compute_blob_kzg_proof(&blob, &commitment, kzg_settings);
+    if let Some(expected_proof) = output {
+        match result {
+            Ok(proof) => {
+                assert_eq!(proof, expected_proof);
+                Ok(())
+            }
+            Err(_) => Ok(()),
+        }
+    } else {
+        let result = compute_blob_kzg_proof(&blob, &commitment, kzg_settings);
+        assert!(result.is_err());
+        Ok(())
+    }
+}
+
+fn run_verify_blob_kzg_proof_test(
+    test: &TestCase,
+    kzg_settings: &KzgSettings,
+) -> Result<(), Error> {
+    let path = &test.data_path;
+    let path = path.to_string() + "/data.yaml";
+    let test_data: serde_yaml::Value = load_yaml(&path);
+    let input_yaml = test_data.get("input").unwrap();
+    let blob_yaml = input_yaml.get("blob").unwrap();
+    let commitment_yaml = input_yaml.get("commitment").unwrap();
+    let proof_yaml = input_yaml.get("proof").unwrap();
+
+    let output_yaml = test_data.get("output").unwrap();
+    let output_result: Result<Option<bool>, _> = serde_yaml::from_value(output_yaml.clone());
+    let output = output_result.unwrap();
+
+    let blob: Blob = match serde_yaml::from_value(blob_yaml.clone()) {
+        Ok(blob) => blob,
+        Err(_) => {
+            assert!(output.is_none());
+            return Ok(());
+        }
+    };
+
+    let commitment = match serde_yaml::from_value(commitment_yaml.clone()) {
+        Ok(commitment) => commitment,
+        Err(_) => {
+            assert!(output.is_none());
+            return Ok(());
+        }
+    };
+
+    let proof = match serde_yaml::from_value(proof_yaml.clone()) {
+        Ok(proof) => proof,
+        Err(_) => {
+            assert!(output.is_none());
+            return Ok(());
+        }
+    };
+
+    let result = verify_blob_kzg_proof(&blob, &commitment, &proof, kzg_settings);
+    if let Some(expected_validity) = output {
+        if expected_validity {
+            assert!(result.is_ok());
+            Ok(())
+        } else {
+            assert!(result.is_err());
+            Ok(())
+        }
+    } else {
+        let result = verify_blob_kzg_proof(&blob, &commitment, &proof, kzg_settings);
+        assert!(result.is_err());
+        Ok(())
+    }
+}
+
+fn run_verify_blob_kzg_proof_batch_test(
+    test: &TestCase,
+    kzg_settings: &KzgSettings,
+) -> Result<(), Error> {
+    let path = &test.data_path;
+    let path = path.to_string() + "/data.yaml";
+    let test_data: serde_yaml::Value = load_yaml(&path);
+    let input_yaml = test_data.get("input").unwrap();
+    let blobs_yaml = input_yaml.get("blobs").unwrap();
+    let commitments_yaml = input_yaml.get("commitments").unwrap();
+    let proofs_yaml = input_yaml.get("proofs").unwrap();
+    let output_yaml = test_data.get("output").unwrap();
+
+    let output_result: Result<Option<bool>, _> = serde_yaml::from_value(output_yaml.clone());
+    let output = output_result.unwrap();
```

Comment:
I think we can just do this for **all** of the outputs

```suggestion
    let output: Option<bool> = serde_yaml::from_value(output_yaml.clone()).unwrap();
```